mods/d2k/rules/infantry.yaml

```diff
@@ -281,11 +281,16 @@ saboteur:
 		Flashes: 0
 		EnterBehaviour: Suicide
 	-RevealOnFire:
+	GrantChargeDrainCondition:
+		DeployedCondition: deployed
+		ChargeTicks: 1250
+		DrainTicks: 375
 	Cloak:
-		InitialDelay: 85
+		RequiresCondition: deployed
+		InitialDelay: 0
 		CloakDelay: 85
 		CloakSound: STEALTH1.WAV
-		UncloakOn: Attack, Unload, Infiltrate, Demolish, Move, Damage, Heal
+		UncloakOn: Attack, Unload, Infiltrate, Demolish, Damage, Heal
```

Comment:
How did the original behave when attacking/unloading/infiltrating/damaged/healed while cloaked?

Reply:
I don't know, but I cleaned this up nonetheless.

Reply:
The behaviour on damage and demolishing still remain to be resolved!

Reply:
@penev92 wanted a reminder.

Reply:
They are now properly set up 👍